### PR TITLE
dev/core#2305 Fix schema differences in MariaDB 10.2.x+

### PR DESCRIPTION
**Pull request by JKingsnorth**
See https://lab.civicrm.org/dev/core/-/issues/2305

Changes in code:

- Change strstr to case-insensitive version to cope with differences in MariaDB
- Add check for NULL/empty default values which are 'deliberate' and can be ignored

### CODE REVIEW

File: CRM/Logging/Schema.php [hunk 1]
```diff
@@ -736,8 +736,10 @@ public function columnsWithDiffSpecs($civiTable, $logTable) {
             // if is-null property is different, and log table's column is NOT-NULL, surely consider the column
             $diff['MODIFY'][] = $col;
           }
-          elseif ($civiTableSpecs[$col]['COLUMN_DEFAULT'] != CRM_Utils_Array::value('COLUMN_DEFAULT', $logTableSpecs[$col]) &&
-            !strstr($civiTableSpecs[$col]['COLUMN_DEFAULT'], 'TIMESTAMP')
+          elseif (
+            $civiTableSpecs[$col]['COLUMN_DEFAULT'] != CRM_Utils_Array::value('COLUMN_DEFAULT', $logTableSpecs[$col])
```
Comment (eileenmcnaughton) [inline CRM/Logging/Schema.php:740]: minor point but when we touch code with CRM_Utils_Array::value we change it to 

```
$logTableSpecs[$col]['COLUMN_DEFAULT'] ?? NULL;
```

**Comment by eileenmcnaughton** [comment]
I've seen this bug so can test this out

**Comment by eileenmcnaughton** [comment]
unrelated test fail but I'll kick off again to get the green tick since I won't come back to this this week

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by JKingsnorth** [comment]
Thanks @eileenmcnaughton - I've quickly updated the description with the minor changes here. Do you think anything else is required here for a merge?

**Comment by eileenmcnaughton** [comment]
@JKingsnorth maybe put the explanation as in-code comments might be more helpful for later coders

**Comment by seamuslee001** [comment]
@JKingsnorth `Parse error: syntax error, unexpected ';' in CRM/Logging/Schema.php on line 740`

**Comment by JKingsnorth** [comment]
> @JKingsnorth `Parse error: syntax error, unexpected ';' in CRM/Logging/Schema.php on line 740`

Yeh, sorry, I fumbled, I pushed before it was ready - I'm on it XD

**Comment by JKingsnorth** [comment]
Also fixing a bug where changing from no column default to a 'falsey' one, ie: 0, is treated as no change =]

**Comment by JKingsnorth** [comment]
OK, this is ready for QA again @eileenmcnaughton / @seamuslee001 - sorry for the noise.

I've amended the comments as per Eileen's suggestion, along with fixing the calls to Utils_Array. It also correctly accounts for 'falsey' default values now (that was originally sloppy on my part, sorry ;] ).

HAPPY NEW YEAR. Oh I'm a bit late on that. Nevermind.

**Comment by eileenmcnaughton** [comment]
@JKingsnorth I don't think you are late - 2020 seems to be hanging around from what I can tell. Discussion at https://lab.civicrm.org/dev/core/-/issues/2020

**Comment by seamuslee001** [comment]
This looks fine to me but haven't r-run this but the code change makes sense to me

**Comment by eileenmcnaughton** [comment]
Thanks @JKingsnorth the upgrade ran MUCH faster with this!

**Pull request merged**